lang/cs/Org.Apache.REEF.Network/Group/Driver/Impl/ContextManager.cs

```diff
@@ -0,0 +1,148 @@
+        private readonly int _totalNumberOfContexts;
+        private readonly IDictionary<string, IActiveContext> _activeContexts;
+
+        public ContextManager(int numberOfContexts)
+        {
+            _totalNumberOfContexts = numberOfContexts;
+            _activeContexts = new Dictionary<string, IActiveContext>();
+        }
+
+        /// <summary>
+        /// Check if the number of active context has reached to the total number
+        /// </summary>
+        /// <returns></returns>
+        public bool AllContextReceived()
+        {
+            lock (_contextLock)
+            {
+                if (_activeContexts.Count < _totalNumberOfContexts)
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Get number of active context in the collection
+        /// </summary>
+        public int Count
+        {
+            get { return _activeContexts.Count; }
+        }
+
+        /// <summary>
+        /// Get all current active context in the collection
+        /// </summary>
+        public ICollection<IActiveContext> ActiveContexts
+        {
+            get
+            {
+                return _activeContexts.Values;
+            }
+        }
+
+        /// <summary>
+        /// Add an active context to the collection
+        /// </summary>
+        /// <param name="context"></param>
+        public void AddContext(IActiveContext context)
+        {
+            _contextLock.EnterWriteLock();
+            try
+            {
+                if (_activeContexts.ContainsKey(context.Id))
+                {
+                    Exceptions.Throw(new Exception("The context is already in the collection" + context.Id), Logger);
+                }
+
+                _activeContexts.Add(context.Id, context);
+            }
+            finally
+            {
+                _contextLock.ExitWriteLock();
+            }
+        }
+
+        /// <summary>
+        /// Remove an active context from the collection 
+        /// </summary>
+        /// <param name="contextId"></param>
+        public void RemoveContext(string contextId)
+        {
+            _contextLock.EnterWriteLock();
+            try
+            {
+                if (_activeContexts.ContainsKey(contextId))
+                {
+                    _activeContexts.Remove(contextId);
+                }
+            }
+            finally
+            {
+               _contextLock.ExitWriteLock(); 
+            }
+        }
+
+        /// <summary>
+        /// Get an active context by id
+        /// </summary>
+        /// <param name="contextId"></param>
+        /// <returns></returns>
+        public IActiveContext GetContext(string contextId)
+        {
+            IActiveContext context = null;
+            _contextLock.EnterReadLock();
+            try
+            {
+                _activeContexts.TryGetValue(contextId, out context);
```

Comment:
There are always two ways to do it, either trow exception here or return null to the caller and let caller to check null if the method cannot find the object for the id they passed in. 
I can add exception here.